### ISSUE TITLE
Optifine crash 1.16.3

### ISSUE THREAD

**Issue by konata2027** [Wurst-Imperium/Wurst7#187]
## Description
Wurst incompatible with Optifine

## Steps To Reproduce
fabric-loader-0.9.3+build.207-1.16.3
fabric-api-0.21.0+build.407-1.16.jar
optifabric-1.6.2.jar
OptiFine_1.16.3_HD_U_G3.jar
Wurst-Client-v7.5-MC1.16.3.jar

## Crash Report
```
---- Minecraft Crash Report ----
// Surprise! Haha. Well, this is awkward.

Time: 9/21/20 10:55 AM
Description: Unexpected error

org.spongepowered.asm.mixin.transformer.throwables.MixinTransformerError: An unexpected critical error was encountered
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:364)
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClass(MixinTransformer.java:191)
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClassBytes(MixinTransformer.java:178)
	at org.spongepowered.asm.mixin.transformer.FabricMixinTransformerProxy.transformClassBytes(FabricMixinTransformerProxy.java:23)
	at net.fabricmc.loader.launch.knot.KnotClassDelegate.getPostMixinClassByteArray(KnotClassDelegate.java:157)
	at net.fabricmc.loader.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:148)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:351)
	at net.minecraft.class_310.method_1523(class_310.java:1014)
	at net.minecraft.class_310.method_1514(class_310.java:656)
	at net.minecraft.client.main.Main.main(Main.java:215)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at net.fabricmc.loader.game.MinecraftGameProvider.launch(MinecraftGameProvider.java:192)
	at net.fabricmc.loader.launch.knot.Knot.init(Knot.java:140)
	at net.fabricmc.loader.launch.knot.KnotClient.main(KnotClient.java:26)
Caused by: org.spongepowered.asm.mixin.injection.throwables.InjectionError: Critical injection failure: Redirector hasStatusEffectApplyFog(Lnet/minecraft/class_1309;Lnet/minecraft/class_1291;)Z in wurst.mixins.json:BackgroundRendererMixin failed injection check, (0/1) succeeded. Scanned 1 target(s). Using refmap Wurst-Client-refmap.json
	at org.spongepowered.asm.mixin.injection.struct.InjectionInfo.postInject(InjectionInfo.java:408)
	at org.spongepowered.asm.mixin.transformer.MixinTargetContext.applyInjections(MixinTargetContext.java:1291)
	at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.applyInjections(MixinApplicatorStandard.java:1042)
	at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.applyMixin(MixinApplicatorStandard.java:395)
	at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.apply(MixinApplicatorStandard.java:320)
	at org.spongepowered.asm.mixin.transformer.TargetClassContext.applyMixins(TargetClassContext.java:345)
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:570)
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:351)
	... 16 more


A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- System Details --
Details:
	Minecraft Version: 1.16.3
	Minecraft Version ID: 1.16.3
	Operating System: Linux (amd64) version 5.8.10-xanmod1
	Java Version: 1.8.0_261, Oracle Corporation
	Java VM Version: Java HotSpot(TM) 64-Bit Server VM (mixed mode), Oracle Corporation
	Memory: 251443816 bytes (239 MB) / 488923136 bytes (466 MB) up to 2134114304 bytes (2035 MB)
	CPUs: 2
	JVM Flags: 5 total; -Xmx2048M -XX:+UseConcMarkSweepGC -XX:+CMSIncrementalMode -XX:-UseAdaptiveSizePolicy -Xmn128M
	Fabric Mods: 
		fabric: Fabric API 0.21.0+build.407-1.16
		fabric-api-base: Fabric API Base 0.1.3+12a8474c95
		fabric-biome-api-v1: Fabric Biome API (v1) 2.0.0+c29459ab95
		fabric-blockrenderlayer-v1: Fabric BlockRenderLayer Registration (v1) 1.1.4+c6a8ea8995
		fabric-command-api-v1: Fabric Command API (v1) 1.0.8+5ce5339895
		fabric-commands-v0: Fabric Commands (v0) 0.2.0+52d3083695
		fabric-containers-v0: Fabric Containers (v0) 0.1.8+045df74f95
		fabric-content-registries-v0: Fabric Content Registries (v0) 0.1.9+059ea86695
		fabric-crash-report-info-v1: Fabric Crash Report Info (v1) 0.1.2+b7f9825d95
		fabric-events-interaction-v0: Fabric Events Interaction (v0) 0.4.0+e2e6cdad95
		fabric-events-lifecycle-v0: Fabric Events Lifecycle (v0) 0.2.0+16acbe5b95
		fabric-game-rule-api-v1: Fabric Game Rule API (v1) 1.0.1+74685c1495
		fabric-item-api-v1: Fabric Item API (v1) 1.1.0+29679fd695
		fabric-item-groups-v0: Fabric Item Groups (v0) 0.2.0+438f963695
		fabric-key-binding-api-v1: Fabric Key Binding API (v1) 1.0.1+730711c695
		fabric-keybindings-v0: Fabric Key Bindings (v0) 0.2.0+3fa9f7c595
		fabric-lifecycle-events-v1: Fabric Lifecycle Events (v1) 1.2.0+74cc3b2095
		fabric-loot-tables-v1: Fabric Loot Tables (v1) 1.0.1+432ea18895
		fabric-mining-levels-v0: Fabric Mining Levels (v0) 0.1.2+b764ce9995
		fabric-models-v0: Fabric Models (v0) 0.1.0+dfdb52d695
		fabric-networking-blockentity-v0: Fabric Networking Block Entity (v0) 0.2.5+b50ffc7b95
		fabric-networking-v0: Fabric Networking (v0) 0.1.10+e00ecb5f95
		fabric-object-builder-api-v1: Fabric Object Builder API (v1) 1.8.0+c1aa8ed895
		fabric-object-builders-v0: Fabric Object Builders (v0) 0.7.0+432ea18895
		fabric-particles-v1: fabric-particles-v1 0.2.2+fb0d9b0e95
		fabric-registry-sync-v0: Fabric Registry Sync (v0) 0.5.0+c1aa8ed895
		fabric-renderer-api-v1: Fabric Renderer API (v1) 0.3.0+8bcfced895
		fabric-renderer-indigo: Fabric Renderer - Indigo 0.4.0+8bcfced895
		fabric-renderer-registries-v1: Fabric Renderer Registries (v1) 2.1.0+e2862de695
		fabric-rendering-data-attachment-v1: Fabric Rendering Data Attachment (v1) 0.1.3+b7f9825d95
		fabric-rendering-fluids-v1: Fabric Rendering Fluids (v1) 0.1.11+fb0d9b0e95
		fabric-rendering-v0: Fabric Rendering (v0) 1.1.0+5341049095
		fabric-rendering-v1: Fabric Rendering (v1) 1.3.0+432ea18895
		fabric-resource-loader-v0: Fabric Resource Loader (v0) 0.3.1+facf3bbf95
		fabric-screen-handler-api-v1: Fabric Screen Handler API (v1) 1.1.0+8724984195
		fabric-structure-api-v1: Fabric Structure API (v1) 1.0.1+fb0d9b0e95
		fabric-tag-extensions-v0: Fabric Tag Extensions (v0) 1.0.3+ac8e8c5995
		fabric-textures-v0: Fabric Textures (v0) 1.0.4+eae12eb895
		fabric-tool-attribute-api-v1: Fabric Tool Attribute API (v1) 1.2.2+ccce563e95
		fabricloader: Fabric Loader 0.9.3+build.207
		minecraft: Minecraft 1.16.3
		mm: Manningham Mills 2.1
		optifabric: OptiFabric 1.6.2
		wurst: Wurst Client v7.5-MC1.16.3
	Launched Version: fabric-loader-0.9.3+build.207-1.16.3
	Backend library: LWJGL version 3.2.2 build 10
	Backend API: AMD TURKS (DRM 2.50.0 / 5.8.10-xanmod1, LLVM 10.0.0) GL version 3.1 Mesa 20.0.8, X.Org
	GL Caps: Using framebuffer using OpenGL 3.0
	Using VBOs: Yes
	Is Modded: Definitely; Client brand changed to 'fabric'
	Type: Client (map_client.txt)
	Graphics mode: fast
	Resource Packs: fabric/fabric, vanilla, fabric/fabric-api-base, file/VanillaTweaks.zip, file/CleanGlass.zip, fabric/fabric-blockrenderlayer-v1, fabric/fabric-containers-v0, fabric/fabric-content-registries-v0, fabric/fabric-crash-report-info-v1, fabric/fabric-events-interaction-v0, fabric/fabric-events-lifecycle-v0, fabric/fabric-game-rule-api-v1, fabric/fabric-item-api-v1, fabric/fabric-item-groups-v0, fabric/fabric-key-binding-api-v1, fabric/fabric-keybindings-v0, fabric/fabric-lifecycle-events-v1, fabric/fabric-loot-tables-v1, fabric/fabric-mining-levels-v0, fabric/fabric-models-v0, fabric/fabric-networking-blockentity-v0, fabric/fabric-networking-v0, fabric/fabric-object-builder-api-v1, fabric/fabric-object-builders-v0, fabric/fabric-registry-sync-v0, fabric/fabric-renderer-api-v1, fabric/fabric-renderer-indigo, fabric/fabric-renderer-registries-v1, fabric/fabric-rendering-data-attachment-v1, fabric/fabric-rendering-fluids-v1, fabric/fabric-rendering-v0, fabric/fabric-rendering-v1, fabric/fabric-resource-loader-v0, fabric/fabric-screen-handler-api-v1, fabric/fabric-tag-extensions-v0, fabric/fabric-textures-v0, fabric/fabric-tool-attribute-api-v1, fabric/fabricloader, fabric/mm, file/SolidSlime.zip, file/VT_Experiment.zip
	Current Language: English (US)
	CPU: 2x Intel(R) Pentium(R) CPU G630 @ 2.70GHz
	OptiFine Version: OptiFine_1.16.3_HD_U_G3
	OptiFine Build: 20200916-082411
	Render Distance Chunks: 10
	Mipmaps: 4
	Anisotropic Filtering: 1
	Antialiasing: 0
	Multitexture: false
	Shaders: null
	OpenGlVersion: 3.1 Mesa 20.0.8
	OpenGlRenderer: AMD TURKS (DRM 2.50.0 / 5.8.10-xanmod1, LLVM 10.0.0)
	OpenGlVendor: X.Org
	CpuCount: 2
```

**Comment by soldtm** [comment]
same issue

**Comment by Frostash** [comment]
so how to fix it or a new update will be made with fixture ?

**Comment by holysnipz** [comment]
might be same as #183 which is claim as fixed

**Comment by boristats** [comment]
same issue

**Comment by T1MOXA** [comment]
> might be same as #183 which is claim as fixed in [0c27d3e](https://github.com/Wurst-Imperium/Wurst7/commit/0c27d3e0adfbc54e2e97dcdd7af2936271bef159)

Thanks, I build it for myself and it works.
Binary download - https://mega.nz/file/0s0z0ATZ#99MdGZUebL2wonCvEFbFryu8U_UaEP6vuhQsOdUgfcU
(For those who don't want to compile sources)

**Comment by Alexander01998** [comment]
This has been fixed in v7.6: https://www.wurstclient.net/updates/wurst-7-6/ (and is the same as #183)
